robottelo/api/utils.py

```diff
@@ -437,3 +438,30 @@ def configure_provisioning(org=None, loc=None):
         'host_group': host_group.name,
         'domain': domain.name,
     }
+
+
+def get_role_by_bz(bz_id):
+    """Create and configure custom role entity for the testing of specific bugs
+     This function will read the dictionary of permissions and their associated
+     bugzilla id's from robottelo.constants "PERMISSIONS_WITH_BZ",
+     these permissions will create filter and a single role will be created
+     from all the filters.
+
+     :param bz_id: This is the bugzilla id that is specified in the
+        PERMISSIONS_WITH_BZ list, all the permissions associated with the bz_id
+        will be fetched and filters will be created
+     :return: A single role entity will be created from all the created filters
+     """
+    role = entities.Role().create()
+    for res_type, perms in PERMISSIONS_WITH_BZ.items():
+        perms_with_bz = [x for x in perms if x.get('bz') is
+                         not None and bz_id in x.get('bz')]
+        if perms_with_bz:
+            permissions = [
+                entities.Permission(
+                    name=perm['name'],
+                ).search()[0]
```

Comment:
The problem when I used the res_type to search was , it returned me with a list the all filters available for that res_type, rather than just one permission, on eliminating the res_type we get expected result.